### PR TITLE
Add establish timeouts

### PR DESCRIPTION
**Pull request by marsella**
Adds timeouts to establish functions, using same methods as in #311.

### CODE REVIEW

File: src/bin/merchant/main.rs
Comment (gijsvl) [inline src/bin/merchant/main.rs:63]: We lost this error in the refactor

File: src/bin/customer/establish.rs
Comment (gijsvl) [inline src/bin/customer/establish.rs:539]: Easy to unit test

**Review by gijsvl — CHANGES_REQUESTED** [review]
Hard to review because of the refactor, but I've tested the configurations with a low timeout and it seems to correctly handle that.
Just two minor comments.

File: src/bin/merchant/main.rs
Comment (marsella) [reply src/bin/merchant/main.rs:63]: Re-added, thank you.

File: src/bin/customer/establish.rs
Comment (marsella) [reply src/bin/customer/establish.rs:539]: Good call, added a test with the failure cases I know about.

File: src/bin/customer/establish.rs
Comment (yaymukund) [inline src/bin/customer/establish.rs:146]: I don't feel that separating this code actually makes it more skimmable in practice. At the code level, `fn originate_contract` can do pretty anything that `run()` can do, so I'd want to read them together to have the full picture. 

In fact, I sort of think we should go the other way - I think even the functions that exist solely to add a `with_timeout()` should become inline `async {...}.with_timeout(...)` blocks to keep the protocol logic in one place.

File: src/bin/customer/establish.rs
Comment (yaymukund) [inline src/bin/customer/establish.rs:295]: `unwrap_or_default()`

**Review by yaymukund — CHANGES_REQUESTED** [review]
Left comments inline

File: src/bin/customer/establish.rs
Comment (marsella) [reply src/bin/customer/establish.rs:146]: Yeah, I can get behind this. The sheer volume of parameters plus the non-reusability of the function makes this a worse solution that I was hoping. I will try adding async blocks instead.

File: src/bin/customer/establish.rs
Comment (marsella) [reply src/bin/customer/establish.rs]: Yes, good idea. I macro'd these because it is essentially the same for `CustomerBalance` and `MerchantBalance`.

**Review by gijsvl — APPROVED** [review]

File: src/bin/customer/establish.rs
Comment (marsella) [reply src/bin/customer/establish.rs]: This is now done!

**Comment by marsella** [comment]
Moved everything from establish back into `async` blocks. Will wait for re-review or confirmation from @yaymukund before merging.

**Review by yaymukund — APPROVED** [review]
Looks good!

**Pull request merged**
